app/models/order.rb

```diff
@@ -263,6 +265,35 @@ def close_direct!(user)
     update_attributes! state: 'closed', updated_by: user
   end
 
+  def send_to_supplier(user)
+    Mailer.order_result_supplier(user, self).deliver_now
+    update!(last_sent_mail: Time.now)
+  end
+
+  def do_end_action
```

Comment:
Does this warrant an exclamation mark? (Since it calls `finish!`, which changes the order).